app/src/main/java/org/kiwix/kiwixmobile/zim_manager/fileselect_view/effects/SendFiles.kt

```diff
@@ -0,0 +1,36 @@
+package org.kiwix.kiwixmobile.zim_manager.fileselect_view.effects
+
+import android.app.Activity
+import android.content.Intent
+import android.net.Uri
+import android.os.Build
+import androidx.core.content.FileProvider
+import org.kiwix.kiwixmobile.BuildConfig
+import org.kiwix.kiwixmobile.zim_manager.fileselect_view.adapter.BooksOnDiskListItem.BookOnDisk
+import org.kiwix.kiwixmobile.zim_manager.local_file_transfer.LocalFileTransferActivity
+
+class SendFiles(private val selectedBooks: List<BookOnDisk>) : SideEffect<Unit> {
+  override fun invokeWith(activity: Activity) {
+    val selectedFileShareIntent =
+      Intent(activity.applicationContext, LocalFileTransferActivity::class.java)
+
+    val selectedFileContentURIs = selectedBooks.mapNotNull {
+      if (Build.VERSION.SDK_INT >= 24) {
+        FileProvider.getUriForFile(
+          activity,
+          BuildConfig.APPLICATION_ID + ".fileprovider",
+          it.file
+        )
+      } else {
+        Uri.fromFile(it.file)
+      }
+    }
+    selectedFileShareIntent.putParcelableArrayListExtra(
+      Intent.EXTRA_STREAM,
```

Comment:
This should be your own string now, probably a constant in LocalFileTransferActivity

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1538/commits/43dee04991a392c38855d20716bb48157f64d82b#diff-4843a9531f06185dcfc85879a4e0d38bR29)

Reply:
I will mark the conversation as resolved but just to share a little hint with you for future endeavours, generally if I pass data to an activity I do it through a specific object some thing along the lines of `ActivityNameParamterObject`, this makes adding data really easy as you don't have to add a new string/object in your bundle, this one Parcelable object contains the data to send to the activity